test/steps_test.go

```diff
@@ -2058,3 +2102,457 @@ func tealCheckDryrun(result string) error {
+	args := []future.Arg{
+		{Name: "", AbiType: firstArgType, Desc: ""},
+		{Name: "", AbiType: secondArgType, Desc: ""},
+	}
+	abiMethod = future.Method{
+		Name:    name,
+		Desc:    "",
+		Args:    args,
+		Returns: future.Return{AbiType: returnType, Desc: ""},
+	}
+	return nil
+}
+
+func createMethodObjectWithArgNames(name, firstArgName, firstArgType, secondArgName, secondArgType, returnType string) error {
+	args := []future.Arg{
+		{Name: firstArgName, AbiType: firstArgType, Desc: ""},
+		{Name: secondArgName, AbiType: secondArgType, Desc: ""},
+	}
+	abiMethod = future.Method{
+		Name:    name,
+		Desc:    "",
+		Args:    args,
+		Returns: future.Return{AbiType: returnType, Desc: ""},
+	}
+	return nil
+}
+
+func createMethodObjectWithDescription(name, nameDesc, firstArgType, firstDesc, secondArgType, secondDesc, returnType string) error {
+	args := []future.Arg{
+		{Name: "", AbiType: firstArgType, Desc: firstDesc},
+		{Name: "", AbiType: secondArgType, Desc: secondDesc},
+	}
+	abiMethod = future.Method{
+		Name:    name,
+		Desc:    nameDesc,
+		Args:    args,
+		Returns: future.Return{AbiType: returnType, Desc: ""},
+	}
+	return nil
+}
+
+func checkTxnCount(givenTxnCount int) error {
+	correctTxnCount := abiMethod.GetTxCountFromMethod()
+	if correctTxnCount != givenTxnCount {
+		return fmt.Errorf("txn count %d != %d", givenTxnCount, correctTxnCount)
+	}
+	return nil
+}
+
+func checkMethodSelector(givenMethodSelector string) error {
+	correctMethodSelector := fmt.Sprintf("%x", abiMethod.GetSelector())
+	if correctMethodSelector != givenMethodSelector {
+		return fmt.Errorf("method selector %s != %s", givenMethodSelector, correctMethodSelector)
+	}
+	return nil
+}
+
+func createInterfaceObject(name string) error {
+	abiInterface = future.Interface{
+		Name:    name,
+		Methods: []future.Method{abiMethod},
+	}
+	return nil
+}
+
+func serializeInterfaceObjectIntoJson() error {
+	abiInterfaceJson, err := json.Marshal(abiInterface)
+	if err != nil {
+		return err
+	}
+
+	abiJsonString = string(abiInterfaceJson)
+	return nil
+}
+
+func createContractObject(name string, appId int) error {
+	abiContract = future.Contract{
+		Name:    name,
+		AppId:   uint64(appId),
+		Methods: []future.Method{abiMethod},
+	}
+	return nil
+}
+
+func serializeContractObjectIntoJson() error {
+	abiContractJson, err := json.Marshal(abiContract)
+	if err != nil {
+		return err
+	}
+
+	abiJsonString = string(abiContractJson)
+	return nil
+}
+
+// equality helper methods
+func checkEqualMethods(method1, method2 future.Method) bool {
+	if method1.Name != method2.Name || method1.Desc != method2.Desc {
+		return false
+	}
+
+	for i, arg := range method1.Args {
+		if arg != method2.Args[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func checkEqualInterfaces(interface1, interface2 future.Interface) bool {
+	if interface1.Name != interface2.Name {
+		return false
+	}
+
+	for i, method := range interface1.Methods {
```

Comment:
It's also worthwhile to verify that the number of methods is the same on interface1 and 2.

Othewrise interface2 could have more extra methods that aren't checked by this function. Same thing for `checkEqualContracts`.

That being said, since this is a test, it would be fine to use reflection to check equality if you wanted to do that instead.